deepwell/migrations/20220906103252_deepwell.sql

```diff
@@ -732,6 +732,186 @@ CREATE TABLE filter (
     UNIQUE (site_id, regex, deleted_at)
 );
 
+--
+-- Forums
+--
+
+-- Groups contain categories, and are site-local.
+CREATE TABLE forum_group (
+    forum_group_id BIGSERIAL PRIMARY KEY,
+    site_id BIGINT NOT NULL REFERENCES site(site_id),
+    created_by BIGINT NOT NULL REFERENCES "user"(user_id),
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+    updated_by BIGINT REFERENCES "user"(user_id),
+    updated_at TIMESTAMP WITH TIME ZONE,
+    deleted_by BIGINT REFERENCES "user"(user_id),
+    deleted_at TIMESTAMP WITH TIME ZONE,
+    name TEXT NOT NULL,
+    description TEXT NOT NULL,
+    visible BOOLEAN NOT NULL DEFAULT true,
+    sort_index INTEGER NOT NULL CHECK (sort_index >= 0),
+    from_wikidot BOOLEAN NOT NULL DEFAULT false,
+
+    UNIQUE (site_id, sort_index),
+    UNIQUE (forum_group_id, site_id),
+    CHECK ((updated_by IS NULL) = (updated_at IS NULL)),
+    CHECK ((deleted_by IS NULL) = (deleted_at IS NULL))
+);
+
+-- Categories belong to a group, and are site-local.
+CREATE TABLE forum_category (
+    forum_category_id BIGSERIAL PRIMARY KEY,
+    forum_group_id BIGINT NOT NULL REFERENCES forum_group(forum_group_id),
+    site_id BIGINT NOT NULL REFERENCES site(site_id),
+    created_by BIGINT NOT NULL REFERENCES "user"(user_id),
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+    updated_by BIGINT REFERENCES "user"(user_id),
+    updated_at TIMESTAMP WITH TIME ZONE,
+    deleted_by BIGINT REFERENCES "user"(user_id),
+    deleted_at TIMESTAMP WITH TIME ZONE,
+    name TEXT NOT NULL,
+    description TEXT NOT NULL,
+    sort_index INTEGER NOT NULL CHECK (sort_index >= 0),
+    from_wikidot BOOLEAN NOT NULL DEFAULT false,
+
+    -- Category settings:
+    max_nest_level SMALLINT CHECK (0 <= max_nest_level AND max_nest_level <= 10),
+    per_page_discussion BOOLEAN DEFAULT false,
+    layout TEXT,
+
+    UNIQUE (forum_group_id, sort_index),
+    UNIQUE (forum_category_id, site_id),
```

Comment:
Is this for an index? I'm not sure what structural constraint this represents, given `forum_category_id` is unique by itself.